### ISSUE TITLE
Aligning on collateral usage

### ISSUE THREAD

**Issue by sternhenri** [filecoin-project/specs#426]
Written by @dignifiedquire and @sternhenri, related to past work on [how power is accounted for]() and meant to help us move forward with speccing [fault handling and collateral](https://github.com/filecoin-project/specs/pull/379) (currently on hold until we're all aligned).

We believe most stakeholders (@anorth, @whyrusleeping, @zixuanzh, @nicola, @decentralion) are aligned on the fact we do not need to slash pledge collateral for storage faults. Summarizing the argument to get any counterargs so we can move forward, esp from @jbenet (who was a bit removed from some of the convos).

**Goals:**
- Protocol security
- Simplicity (namely, if possible, separation of concerns wrt collateral)

**Assumptions:**
- goal of storage collateral: is to secure client based stored data
- goal of pledge collateral: is to secure consensus

**Solution:**
- Reminder: for miner M, round B, Power_M(B) = Power_M(last submitted PoSt) / TotalNetworkPower(B-1)
- Proposed changes:
    - laid out [here](https://github.com/filecoin-project/specs/blob/5e43b27e597691de3f07b6d51644dd18397dbce3/faults.md), namely storage collateral is slashed when sectors are not proven in time (a storage fault); pledge collateral is only slashed for consensus faults, triggered when there is proven adversarial behavior in consensus.
- Remaining changes tbd:
    - Add minimum storage collateral & burn parts of it on faults

Why is not necessary to slash pledge collateral, for storage faults?
> Failing to store sealed data undermines consensus security. This is what the pledge is securing in the first place (the power/influence over consensus). 
> @jbenet

*Claim:* Faulting on storage does not undermine consensus security (with the construction as of today)
- Consensus power is directly derived from proven storage with a delay, rather than claimed storage.

*Claim:* The delay does not negatively effect consensus in the case of a miner faulting on their storage.
Simply put, whilst miners can continue to participate in EC for up to 1 Proving Period after dropping sectors, their power is only updated after 1 proving period when they commit storage. Thus, overall, the protocol fairly accounts for storage in consensus, independent of faults.

```
              ┌─────────────┐    ┌─────────────┐  ┌─────────────┐  ┌─────────────────┐
              │Commit Sector│    │    PoSt     │  │    PoSt     │  │  Drop   Fault   │
              └─────────────┘    └─────────────┘  └─────────────┘  └─────────────────┘
               ────────────────▶ ────────────────▶ ────────────────▶◀──────── ────────▶
┌─────────────┐      ┌────┐             ┌────┐           ┌────┐             ┌ ─ ─
│ Secs Used  │      │ 1  │◀─────┐      │ 1  │◀────┐     │ 1  │◀─────┐        0  │◀────┐
└─────────────┘      └────┘      │      └────┘     │     └────┘      │      └ ─ ─      │
                                 │                 │                 │                 │
┌─────────────┐      ┌ ─ ─       │      ┌────┐     │     ┌────┐      │      ┌────┐     │      ┌ ─ ─
|    Power    │        0  │      └──────│ 1  │     └─────│ 1  │      └──────│ 1  │     └──────  0  │
└─────────────┘      └ ─ ─              └────┘           └────┘             └────┘            └ ─ ─
```

Three points worth noting (all true based on current spec):
1. A given sector-set will be in use for the entirety of the ProvingPeriod before commit sector can be called.(**)
1. One can indeed mine with power derived from dropped sectors for up to one Proving Period. On the previous point, it doesn't impact the fact that on the whole, they earn power for as many Proving Periods as they have put sectors to work.
1. If in between PoSt_2 and PoSt_3, a miner decides to commit additional sectors, the extra power will kick in at PoSt_4 (and not PoSt_3), thereby maintaining the invariant laid out above.

(*) The reason for this is that sectors only get added to the proving set at the end of a proving period. As such power for a newly commited sector will only kick in at the end of the proving period following the proving period in which the sector was commited. (In that sense, the "Commit Sector" and first "PoSt" in the diagram above could not be in the same Proving Period).

**Comment by anorth** [comment]
I infer that this issue is primarily for discussion about the use of pledge collateral so I will refrain from voicing other concerns, but...

> Proposed changes:
> laid out here, namely storage collateral is slashed when sectors are not proven in time (a storage fault); pledge collateral is only slashed for consensus faults, triggered when there is proven adversarial behavior in consensus.

... the words in linked commit do not match what is summarised, and still refer to slashing pledge collateral.

**Comment by sternhenri** [comment]
Ack @anorth. You are right, this was changed by @whyrusleeping in 5e43b27e597691de3f07b6d51644dd18397dbce3 before we got to sync (and land where this summary describes). My sense is @dignifiedquire wanted to leave as is until we had full alignment as a team (point of this issue). That said, we should indeed have PR fit intent. Are we ok changing back to fit the intent per our description above, @dignifiedquire (assuming I've not mischaracterized stuff)?

**Comment by taoshengshi** [comment]
> **Assumptions:**
> 
> * goal of storage collateral: is to secure client based stored data
> * goal of pledge collateral: is to secure consensus
> 
Very good overall illustration, we need this kind of high level design no matter whether it is from programmer or user perspective. 

Base on Assumptions 1, storage collateral has close relation with payment channel among contracting parties
> * client consume storage and pay FIL 
> * miner provide storage  and get FIL 
> * client and miner settle accounts on chain
> * any side of the payment channel who breach contracts should be slashed
> * miner fulfill obligations should get payment
So storage collateral should not impact consensus security.

Assumptions 2 is obviously false, pledge collateral is kind of get  "tax"  from all miners to mitigate sybil attacks. and as @sternhenri mentioned `their power is only updated after 1 proving period when they commit storage. `, PoSt is somehow doing the some kind of `PoW` work, so pledge collateral should not impact consensus security too.

To sum up，collateral is economic incentive mechanism rather than some factors impact consensus. @sternhenri correct me if I were wrong.

**Comment by dignifiedquire** [comment]
>  That said, we should indeed have PR fit intent. Are we ok changing back to fit the intent per our description above, @dignifiedquire (assuming I've not mischaracterized stuff)?

Correct, I have reverted the commit from why, to match the intent of the PR
